### ISSUE TITLE
TextInput issue on Android 13 devices

### ISSUE THREAD

**Issue by DanStevensonDev** [FaridSafi/react-native-gifted-chat#2306]
Hi, I've noticed an issue with TextInputs on Android 13 devices, where the input will lag significantly, especially for long text inputs and when deleting a lot of text by press/holding backspace. 

The underlying issue and a (not perfect, but seemingly working) solution is discussed here: https://github.com/facebook/react-native/issues/35155#issuecomment-1303994598

I've put in place the suggested fix locally using `patch-package`. Although I know it's not an issue with `react-native-gifted-chat`, this seems to fix the issue in my app.

I've not been able to test this extensively yet, but posting in case it's useful to others. 

```diff
diff --git a/node_modules/react-native-gifted-chat/lib/Composer.js b/node_modules/react-native-gifted-chat/lib/Composer.js
index 3331d31..f92956f 100644
--- a/node_modules/react-native-gifted-chat/lib/Composer.js
+++ b/node_modules/react-native-gifted-chat/lib/Composer.js
@@ -49,6 +49,8 @@ export default class Composer extends React.Component {
         this.onChangeText = (text) => {
             this.props.onTextChanged(text);
         };
+        // Fix issue with lagging TextInput on Andoid 13 discussed here: https://github.com/facebook/react-native/issues/35155#issuecomment-1303994598
+        this.isAndroid13 = Platform.OS === 'android' && Platform?.constants?.Release === '13'
     }
     render() {
         return (<TextInput testID={this.props.placeholder} accessible accessibilityLabel={this.props.placeholder} placeholder={this.props.placeholder} placeholderTextColor={this.props.placeholderTextColor} multiline={this.props.multiline} editable={!this.props.disableComposer} onChange={this.onContentSizeChange} onContentSizeChange={this.onContentSizeChange} onChangeText={this.onChangeText} style={[
@@ -64,7 +66,7 @@ export default class Composer extends React.Component {
                     },
                 }),
             },
-        ]} autoFocus={this.props.textInputAutoFocus} value={this.props.text} enablesReturnKeyAutomatically underlineColorAndroid='transparent' keyboardAppearance={this.props.keyboardAppearance} {...this.props.textInputProps}/>);
+        ]} autoFocus={this.props.textInputAutoFocus} value={this.props.text} enablesReturnKeyAutomatically underlineColorAndroid='transparent' keyboardAppearance={this.props.keyboardAppearance} {...this.props.textInputProps} keyboardType={this.isAndroid13 ? 'visible-password' : 'default'} />);
     }
 }
 Composer.defaultProps = {
```

**Comment by fukemy** [comment]
did you using props text/setText in your screen? This will cause screen re-render on every text change

**Comment by ahnafnafee** [comment]
`'visible-password'` disables certain keyboard features like Clipboard unfortuntately

**Comment by levepic** [comment]
Still having ANRs with 'visible-password'

**Comment by fikrielhabib** [comment]
i solve the problem from here:
https://github.com/facebook/react-native/issues/30034#issuecomment-806396274

**Comment by levepic** [comment]
Although this is easy to pass as textInputProps this messes up scrolling (always scrolls to top of the window to the oldest messages)

> i solve the problem from here: [facebook/react-native#30034 (comment)](https://github.com/facebook/react-native/issues/30034#issuecomment-806396274)

**Comment by levepic** [comment]
Modifying scrollto in MessageContainer solves the scrolling problem

`this.scrollToBottom = (animated = true) => {
            const { inverted } = this.props;
            if (inverted) {
                this.scrollTo({ offset: 0, animated });
            }
            else if (this.props.forwardRef && this.props.forwardRef.current) {
                **this.scrollTo({ offset: 0, animated });
                //this.props.forwardRef.current.scrollToEnd({ animated });**
            }
        };`

**Comment by fkiDev** [comment]
It works fine with me without modifying MessageContainer. what inverted value do you set for gifted chat props? for me I leave the default value which is true

**Comment by levepic** [comment]
Do you mean its not the "inverted" prop but the scale being set as "transform" that makes the ANRs? So the solution is to remove transform?

**Comment by fkiDev** [comment]
> Do you mean its not the "inverted" prop but the scale being set as "transform" that makes the ANRs? So the solution is to remove transform?

For me if i set inverted on giftedChat props to false without patch on node_modules it works and problem on android 13 goes away but message list has been reversed (that is not what i want). So I leave inverted to true and put the patch I mentioned earlier on node_modules and it works fine

**Comment by levepic** [comment]
So I have added your suggested fix and also added the fix to make ScaleY still available on RN 0.70 in index.js

`// Add scaleY back to work around its removal in React Native 0.70.
import ViewReactNativeStyleAttributes from 'react-native/Libraries/Components/View/ReactNativeStyleAttributes'
ViewReactNativeStyleAttributes.scaleY = true`

and the ANRs are gone from production! Finally fixed it. Thank you!

> i solve the problem from here: [facebook/react-native#30034 (comment)](https://github.com/facebook/react-native/issues/30034#issuecomment-806396274)

**Comment by phatmann** [comment]
The root cause of this issue is a bug in [Samsung's Grammarly integration in Android](https://github.com/facebook/react-native/issues/35590). The `scaleY` trick works for some users but not all, as we confirmed when we deployed it. 

Another suggested workaround in that issue is turning off auto-correct by hacking keyboard type. But that seems a bit too severe as a go-to fix.

**Comment by Johan-dutoit** [comment]
The `scaleY` trick and the input lag are two completely different issues.  I'd highly recommend implementing the `scaleY` fix for Android (performance is poor otherwise).  The input lag is something we have to wait on from Samsung/React Native.

**Comment by efstathiosntonas** [comment]
Probably fixed on 0.68.6, 0.69.8 and at 0.70.x by the end of this week

**Comment by phatmann** [comment]
Samsung recently released an update that seems to fix the input lag when Grammarly integration is on.

**Comment by phatmann** [comment]
@efstathiosntonas is there an open issue or PR in the React Native repo we can check out?

**Comment by efstathiosntonas** [comment]
@phatmann https://github.com/facebook/react-native/issues/35936#issue-1553873168

**Comment by phatmann** [comment]
@efstathiosntonas Ah, that is fix for the text input issue, which makes sense given the subject of this bug. But I was hoping for a fix for the `inverted` issue, since that has been present for years and recently got worse with Android 13.

**Comment by savithrielangovan** [comment]
> `'visible-password'` disables certain keyboard features like Clipboard unfortunately

visible-password also hide keypad emoji's

**Comment by fukemy** [comment]
> Probably fixed on 0.68.6, 0.69.8 and at 0.70.x by the end of this week

Hi, where do you put this code? Thanks
